srcs/cpp/src/quiver/cuda/quiver_sample.cu

```diff
@@ -289,61 +289,61 @@ class TorchQuiver
         }
         return std::make_tuple(out_vertices, row_idx, col_idx);
     }
-};
-
-std::tuple<torch::Tensor, torch::Tensor, torch::Tensor>
-reindex_single(torch::Tensor inputs, torch::Tensor outputs, torch::Tensor count)
-{
-    using T = int64_t;
-    cudaStream_t stream = 0;
-    const auto policy = thrust::cuda::par.on(stream);
-    thrust::device_vector<T> total_inputs(inputs.size(0));
-    thrust::device_vector<T> total_outputs(outputs.size(0));
-    thrust::device_vector<T> input_prefix(inputs.size(0));
-    const T *ptr;
-    size_t bs;
-    ptr = count.data_ptr<T>();
-    bs = inputs.size(0);
-    thrust::copy(ptr, ptr + bs, input_prefix.begin());
-    ptr = inputs.data_ptr<T>();
-    thrust::copy(ptr, ptr + bs, total_inputs.begin());
-    thrust::exclusive_scan(policy, input_prefix.begin(), input_prefix.end(),
-                           input_prefix.begin());
-    ptr = outputs.data_ptr<T>();
-    bs = outputs.size(0);
-    thrust::copy(ptr, ptr + bs, total_outputs.begin());
-
-    const size_t m = inputs.size(0);
-    using it = thrust::counting_iterator<T>;
-
-    thrust::device_vector<T> subset;
-    TorchQuiver::reindex_kernel(stream, total_inputs, total_outputs, subset);
-
-    int tot = total_outputs.size();
-    torch::Tensor out_vertices = torch::empty(subset.size(), inputs.options());
-    torch::Tensor row_idx = torch::empty(tot, inputs.options());
-    torch::Tensor col_idx = torch::empty(tot, inputs.options());
+    std::tuple<torch::Tensor, torch::Tensor, torch::Tensor>
+    reindex_single(torch::Tensor inputs, torch::Tensor outputs, torch::Tensor count)
```

Comment:
I think `reindex` is a stateless function, so at least it is a static class method, or a function, not a non-static class method